seek_to.h

```diff
@@ -0,0 +1,25 @@
+
+#include "leveldb/c.h"
+#include <stdbool.h>
+
+typedef struct {
+    unsigned char valid;
+    unsigned char equal;
+} SeekResult;
+
+SeekResult leveldb_iter_seek_to(leveldb_iterator_t* iter, const char* k, size_t klen) {
+    // do the seek
+    leveldb_iter_seek(iter, k, klen);
+
+    SeekResult sr;
+    sr.valid = leveldb_iter_valid(iter);
+
+    if (sr.valid != 0) {
+        size_t olen;
+        const char* ok = leveldb_iter_key(iter, &olen);
```

Comment:
assuming this is idiom in leveldb code but `ok` is a confusing var name for this give how it's used in go